helix-core/src/main/java/org/apache/helix/controller/rebalancer/topology/Topology.java

```diff
@@ -57,8 +58,19 @@ public enum Types {
   private final Map<String, InstanceConfig> _instanceConfigMap;
   private final ClusterTopologyConfig _clusterTopologyConfig;
 
+  /**
+   * Create a Topology for a cluster.
+   *
+   * @param allNodes           allNodes of the given cluster.
+   * @param liveNodes          liveNodes of the given cluster.
+   * @param instanceConfigMap  instanceConfigMap of the given cluster.
+   * @param clusterConfig      clusterConfig of the given cluster.
+   * @param faultZoneLevelOnly whether to include additional non-faultZone level nodes in the
+   *                           topology tree above the end-nodes.
+   */
   public Topology(final List<String> allNodes, final List<String> liveNodes,
-      final Map<String, InstanceConfig> instanceConfigMap, ClusterConfig clusterConfig) {
+      final Map<String, InstanceConfig> instanceConfigMap, ClusterConfig clusterConfig,
+      boolean faultZoneLevelOnly) {
```

Comment:
Just curious, Please correct me if I cam wrong, I did not find anywhere we are using the default constructor. We always set `faultZoneLevelOnly` to true?

Reply:
Yes, that is correct; however, this constructor is public and may be used outside our code base by others. 

This will keep backwards compatibility, while only changing the behavior of the rebalancer to limit the backwards incompatibility concerns.

Reply:
Yes, we should always keep the default constructor.